### PR TITLE
Write code for swerve drivetrain

### LINKED ISSUE #3: Swerve Drivetrain

### PR DESCRIPTION
**Pull request by a1cd**
closes #3

### CODE REVIEW

File: src/main/java/frc/robot/Constants.java [hunk 1]
```diff
@@ -15,9 +15,10 @@
  * wherever the constants are needed, to reduce verbosity.
  */
 public final class Constants {
-  public static final Mode currentMode = Mode.SIM;
+    public static final Mode currentMode = Mode.SIM;
```
Comment (GBKP) [inline src/main/java/frc/robot/Constants.java]:
```suggestion
    public static final Mode currentMode = Mode.REAL;
```

File: src/main/java/frc/robot/RobotContainer.java [hunk 2]
```diff
@@ -32,73 +32,120 @@
...
+                        new ModuleIOFalcon500(BLTurnMotorId, BLDriveMotorId, false, false, BLTurnEncoderId),
+                        new ModuleIOFalcon500(FLTurnMotorId, FLDriveMotorId, false, false, FLTurnEncoderId),
+                        new ModuleIOFalcon500(BRTurnMotorId, BRDriveMotorId, false, false, BRTurnEncoderId),
+                        new GyroIOReal(gyroID, "rio"),
+                        new VisionIOPhoton());
+//                flywheel = new Flywheel(new FlywheelIOSparkMax());
+                break;
+
+            // Sim robot, instantiate physics sim IO implementations
+            case SIM:
```
Comment (GBKP) [inline src/main/java/frc/robot/RobotContainer.java:83]: Pretty sure this is right but just making sure this shouldn't be real I think you got it right here though

File: src/main/java/frc/robot/subsystems/drive/Drive.java [hunk 3]
```diff
@@ -1,74 +1,285 @@
...
+        }
+
+
+        angleFF = new SimpleMotorFeedforward(.25928, .28217, .0050137);
+
+
+        drivePIDs = new ArrayList<>(4);
+        for (int i = 0; i < 4; i++) {
+            drivePIDs.add(new ProfiledPIDController(
+                    2.37,
```
Comment (GBKP) [inline src/main/java/frc/robot/subsystems/drive/Drive.java:114]: shouldnt the proportion be less than that?

**Review by GBKP — COMMENTED** [review]
nice

File: src/main/java/frc/robot/subsystems/drive/Drive.java (hunk 3 above)
Comment (a1cd) [reply src/main/java/frc/robot/subsystems/drive/Drive.java:114]: Still tuning pis

Comment (a1cd) [reply src/main/java/frc/robot/subsystems/drive/Drive.java:114]: PID

**Review by GBKP — APPROVED** [review]

**Pull request merged**